src/cls/rbd/cls_rbd.cc

```diff
@@ -2712,12 +2714,17 @@ int object_map_snap_remove(cls_method_context_t hctx, bufferlist *in,
   }
 
   bool updated = false;
-  for (uint64_t i = 0; i < dst_object_map.size(); ++i) {
-    if (dst_object_map[i] == OBJECT_EXISTS_CLEAN &&
-        (i >= src_object_map.size() || src_object_map[i] == OBJECT_EXISTS)) {
-      dst_object_map[i] = OBJECT_EXISTS;
+  auto src_it = src_object_map.begin();
+  auto dst_it = dst_object_map.begin();
+  auto dst_it_end = dst_object_map.end();
+  uint64_t i = 0;
+  for (; dst_it != dst_it_end; ++dst_it, ++src_it) {
```

Comment:
Cannot increment ```src_it``` if ```i``` is beyond the original object map size

Reply:
@dillaman  sorry for late to response,  i committed codes to fix this, please take a review again. thanks